### PR TITLE
(Frontend) Adds Age Adjustment Card

### LINKED ISSUE #966: Age adjusting COVID data
We should likely do this, question is how much work is it & how we should prioritize

### PR DESCRIPTION
**Pull request by benhammondmusic**
## Preview Link

<a href="https://deploy-preview-1459--health-equity-tracker.netlify.app/exploredata?dt1=deaths&mls=1.covid-3.06#age-adjusted" target="_blank">
  <br />
Click to demo updates </a>  

## To Do
- [x] troubleshoot data source footer; not showing CDC restricted ? 
- [x] briefly address age-adjustment on the methodology page, with link to new age-adj tab
- [x] suggestion from CDC meeting was to consider showing the non-age adjusted ratio alongside the age-adjusted ratio, allowing the user to see how the adjustment changed the disparity. (Ben note: they do not present this data, and josh noted we would be comparing different population sources this way.... this should likely be a "consider for later discussion" feature and not part of the launch)
- [x] once low numbers are sent as `null` from backend, look at overriding the `!` message from `No data available` to `insufficient data`. Should this be site-wide or just for the age-adj table?
- [x] CDC suggestion to round to one decimal point (on backend)
- [x] CDC suggestion to display very small numbers as simply `< 0.1x` or similar. (from Ben: I feel like our `!` icon accomplishes this in a less specific and less confusing way and would like to consider staying with that rather than introducing a new technique for showing too-small data)
- [x] hide table when sex toggle is selected
- [x] when sex toggle is selected, offer missing data alert. 
- [x] Possibly offer link to different filter w age adjustment available (by race)
- [x] ensure correct population source is present (hard coded for now since only select variables are actually age-adj)
- [x] remove json tmp files and env line
- [x] update methodology copy https://github.com/SatcherInstitute/health-equity-tracker/issues/966#issuecomment-901283431
- [x] write small info alert text explaining what age-adjustment is and why it is important for health equity
- [x] for data types that dont have age-adjustments but we do have the data for other data types (eg covid cases) present age-adjusted card with no table and only the above info box, dynamically linking to the available age-adj data types
- [x] finish MVP card design
- [x] hook up "missing data" alert that either states age adjustment ISNT APPROPRIATE or ISNT AVAILABLE YET
- [x] connect incoming data from backend
- [x] connect via the back of the frontend (providers, etc)
- [x] conditionally render only for appropriate tracker settings
- [x] wording for INFO BOX (why it's important)
- [x] wording for NOT APPLICABLE ALERT (why it's not applicable)
- [x] detailed wording for METHODOLOGY (specifically how)
- [x] We need to decide which conditions / data types would be appropriate for age-adjustment. Emailed Ebony 2/22
- [x] Confirm national number are working (currently not in attached `tmp/` data json)
- [x] update tests? Since calculations are all done on the backend, is there any need for frontend testing of age-adjustment?  to expect correct ratio value; either `null` if appropriate but not yet available, `undefined` if not appropriate or expected in the future, or a `float` value (depending on the condition data type). Currently only 3 types of tests are implemented for this provider: County/Race; State/Age; and National/Sex. We would need to State or National / Race to test this new feature

## Description
Adds a new card displaying age-adjusted ratios

- COVID Deaths and Hospitalizations (not cases)
- By Race 
- National and State level reports

## Motivation and Context

closes #966 

filed new issue for expanding age-adjustments to other conditions #1467 

## Types of changes

- New feature (non-breaking change which adds functionality)

### CODE REVIEW

**Comment by joshzarrabi** [comment]
I'm gonna put this in draft mode till the backend is approved

**Comment by benhammondmusic** [comment]
currently injecting fake data in the provider step; component seems to be rendering properly

**Comment by benhammondmusic** [comment]
Screen Shot 2022-02-17 at 12 40 23 PM

**Comment by benhammondmusic** [comment]
Screen Shot 2022-02-17 at 2 28 55 PM

**Comment by benhammondmusic** [comment]
We need to decide what, if anything, to display on other tracker settings that don’t currently or won’t ever have age-adjusted ratios to display.

- Option 1: Render AGE ADJUSTED CARD AND TABLE only on specific tracker settings including condition, data type, demographic, geography. 

PROs: Less noise on the page for conditions that don’t / won’t have age adjustments
CONs: This is an important metric to display, and since the tracker defaults to covid cases, many if not most people will never see this card. There is a separate concern that our data type and demographic toggles are too hard to find and should be made more prominent. 

- Option 2: Render AGE ADJUSTED CARD on all conditions where age is an underlying factor (ie where age-adjustment could be appropriate), and display actual AGE ADJUSTED TABLE for exact toggle settings where it exists, and where it doesn’t exist provide a “Missing data” style ALERT that explains how age-adjustment would be helpful for health equity but we don’t have it, and links to alternate data type + demographic settings where age-adjusted number are available

PROS: reduces confusion by keeping table off of conditions where it isn’t relevant; provides context for conditions where we don’t have the adjustments but they might be helpful; good UX to allow user to understand that type of data exists on alternate settings and provide easy navigation to said settings
CONS: We don’t currently hide any cards based on the condition, so it might be confusing to the user and won’t match our existing pattern. The compare view might leave a large gap if one condition offers age-adj and the other doesn’t

- Option 3: Always display the CARD, with a dynamic ALERT if the TABLE isn’t available that explains we a) don’t have the age-adjusted data or b) age-adjustment isn’t helpful for this condition

PROS: matches current pattern; alerts user to the existence of this analysis on every tracker view
CONS: more stuff on the page; most conditions won’t have age-adjustment so it’s a lot of extra “noise” potentially

**Comment by benhammondmusic** [comment]
## Proceeding with Option 3 outlined above. 

> Option 3: Always display the CARD, with a dynamic ALERT if the TABLE isn’t available that explains we a) don’t have the age-adjusted data or b) age-adjustment isn’t helpful for this condition
> PROS: matches current pattern; alerts user to the existence of this analysis on every tracker view
> CONS: more stuff on the page; most conditions won’t have age-adjustment so it’s a lot of extra “noise” potentially

We need to decide:
- [ ] which condition data types would be appropriate for age-adjustment
- [ ] finalize the alert warning for the Not Applicable Alert. 

Age-Adjusted:
- Covid Deaths
- Covid Hosp

Not Yet Age Adjusted (but appropriate if data were available):
- Diabetes? 
- Chronic Kidney Disease?
- Cardiovascular Disease?
- COPD?
- Asthma?

Age Adjustment N/A:
- Covid cases
- covid vaccinations
- voter participation
- care avoidance?
- preventable hospitalizations? 
- depression? 
- excessive drinking?
- frequent mental distress?
- suicide?
- illicit opioid / non-rx opioid / substance misuse
- poverty?
- uninsurance?


PLACEHOLDER ALERT WORDING FOR AGE-ADJUSTMENT N/A
`We do not plan to calculate age-adjusted ratios for Voter Participation.`

**Comment by jgonzalezmsm** [comment]
Consider:

**PLACEHOLDER ALERT WORDING FOR AGE-ADJUSTMENT N/A**
Age Adjustment is a statistical process applied to rates of disease, death, or other health outcomes that occur more frequently among different age groups. Adjusting for age allows for fairer comparison between populations, however it is not needed when age is not a risk-factor or correlated to outcomes. As a result, we do not age adjust for {voter participation}.

**Comment by benhammondmusic** [comment]
Per Josh: I would like to say the sentence: "these are hypothetical deaths rates that would be if every racial group had the same age breakdown as white people" or something;

I think thats more clear than "normalize by age"

**Comment by benhammondmusic** [comment]
Side by side view of both types of tables (Data and newer Age Adjusted). I made the data table more blue and the age-adjusted more green. Open to feedback on color choices... should they be more bold and distinct? 

Screen Shot 2022-02-23 at 1 42 21 PM

**Comment by joshzarrabi** [comment]
> Side by side view of both types of tables (Data and newer Age Adjusted). I made the data table more blue and the age-adjusted more green. Open to feedback on color choices... should they be more bold and distinct?
> 

Looks really nice. I think we should explain why we chose not to age adjust.

`Because outcomes for <condition> are not heavily influenced by age, we chose not to provide age adjusted numbers.`?

**Comment by benhammondmusic** [comment]
> > Side by side view of both types of tables (Data and newer Age Adjusted). I made the data table more blue and the age-adjusted more green. Open to feedback on color choices... should they be more bold and distinct?
> 
> Looks really nice. I think we should explain why we chose not to age adjust.
> 
> `Because outcomes for <condition> are not heavily influenced by age, we chose not to provide age adjusted numbers.`?

Definitely; I like that, it's short but succinct. And then maybe merge what JC wrote before and what Vansh had written and use that in the INFO ALERT that appears on every card (regardless of the data being there or not)

**Comment by benhammondmusic** [comment]
Note: we should be consistent and use `age-adjusted` with the hyphen; that seems to be the standardized spelling

**Comment by benhammondmusic** [comment]
4 Different Views:

## COVID-19 Deaths (available)
Screen Shot 2022-02-23 at 6 32 51 PM

## COVID-19 Cases (not available; alternate data types offered)
Screen Shot 2022-02-23 at 6 32 24 PM

## ASTHMA (Missing Data)
Screen Shot 2022-02-23 at 6 31 59 PM

## VOTER PARTICIPATION (N/A)
Screen Shot 2022-02-23 at 6 33 37 PM

**Comment by jgonzalezmsm** [comment]
All look great, would just like to clean up the info box text, will work on that in the morning

**Comment by joshzarrabi** [comment]
It looks beautiful

**Comment by benhammondmusic** [comment]
Thanks guys. Also, we should figure out phrasing to not confuse causation and correlation. Like, older people are more likely to vote, but that's still not a good reason to age-adjust, right?

**Comment by benhammondmusic** [comment]
## Filter out `White` as it will always be 1, and `Two or More & Unrepresented` since we will not by providing that metric on the backend

Screen Shot 2022-02-28 at 2 48 09 PM

**Comment by joshzarrabi** [comment]
Nice,  We still have to change the population source to the other one.

I'll work on adding that

**Comment by joshzarrabi** [comment]
Also, it looks like states that have 0 deaths are still showing up on the map, I need to make sure they are properly filtered out of the backend.

**Comment by benhammondmusic** [comment]
Screen Shot 2022-02-28 at 2 55 56 PM

**Comment by benhammondmusic** [comment]
Screen Shot 2022-03-01 at 12 11 54 PM

**Comment by jgonzalezmsm** [comment]
**Proposed Info Box Text - With Data**

Age Adjustment is a statistical process applied to rates of disease, death, or other health outcomes that occur more frequently among different age groups. Adjusting for age allows for fairer comparison between populations, where age is a large risk factor. By computing rates that are normalized for age, we can paint a more accurate picture of undue burden of disease and death between populations. The ratio in the table below indicates the increased likelihood of death from Covid-19 for the listed racial group. More details can be found on our `Methodology Page`.

**Proposed Info Box Text - Age Adjustment Would be Helpful, but not yet available**

Age Adjustment is a statistical process applied to rates of disease, death, or other health outcomes that occur more frequently among different age groups. Adjusting for age allows for fairer comparison between populations, where age is a large risk factor. This process relies on publicly available data to derive the necessary computations, however data on `Topic` is limited or unavailable. `Read more on the impact of Missing Data.`

**Proposed info Box Text - Age Adjustment Not Needed**

Age Adjustment is a statistical process applied to rates of disease, death, or other health outcomes that occur more frequently among different age groups. Adjusting for age allows for fairer comparison between populations, however it is not needed when age is not a risk-factor or correlated to outcomes. As a result, we do not age adjust for {voter participation}.

**Methodology Page**

pending - combine CDC (cite it) with our specific process, ensure disparate impact / health equity lens

**Comment by benhammondmusic** [comment]
https://docs.google.com/document/d/1kyU7nZpWfEH1rhYVGvwb5Y7UeQQAqhzIgySWoNswN-c/edit

**Comment by jgonzalezmsm** [comment]
Ebony proposal:

Proposed info Box Text - Age Adjustment Not Needed

Age Adjustment is a statistical process applied to rates of disease, death, or other health outcomes that occur more frequently among different age groups. Adjusting for age allows for fairer comparison between populations, **however we do not age adjust <topic> on the Health Equity Tracker but our source data may adjust when compiling ... distinguish if source data does age adjust.**

^ JC to clean that up.

If no data on age adjustment, serve our regular missing data alert.

**Comment by jgonzalezmsm** [comment]
Josh working on write up (related to CDC meeting), can include on Methodology page.

**Comment by benhammondmusic** [comment]
Updated text. Card will display:
- Always an info box
- Either a data table or a missing data alert

The missing data alert will offer alternate age-adjusted data types by link if they exist and it's on an appropriate demographic type (race or age, not sex)

Screen Shot 2022-03-08 at 1 10 16 PM
Screen Shot 2022-03-08 at 1 12 35 PM
Screen Shot 2022-03-08 at 1 12 55 PM
Screen Shot 2022-03-08 at 1 13 10 PM
Screen Shot 2022-03-08 at 1 13 26 PM
Screen Shot 2022-03-08 at 1 13 49 PM

**Comment by benhammondmusic** [comment]
Translated @joshzarrabi 's markdown doc into html. Here are screenshot or you can view the preview above to see it. Still some cleanup needed

Screen Shot 2022-03-10 at 5 25 49 PM
Screen Shot 2022-03-10 at 5 26 01 PM

**Comment by joshzarrabi** [comment]
@benhammondmusic Looks so nice! Do you think we should make it its own page?

**Comment by benhammondmusic** [comment]
> @benhammondmusic Looks so nice! Do you think we should make it its own page?

Thanks! Realized I missed a few copy pastes on your tables. And yes, I think making a new tab on the data/methodology page would be best; it'll have its own route like `healthequitytracker.org/ageadjustment`

**Comment by jgonzalezmsm** [comment]
really detailed and looks great. thank you both

**Comment by benhammondmusic** [comment]
Merged #1504 into this PR and got everything hooked up for demos. The scroll position can be a little off depending on the screen size; it's worst on mobile. But that's more a problem with our calculations of placeholder card size on mobile then it is this actual feature. Later, we can shorten the "pause before scroll" time, once our #1474 work is done and the app doesn't choke


https://user-images.githubusercontent.com/41567007/159600706-23fcd0ca-a48e-4eb2-b42b-243472445182.mov

**Comment by benhammondmusic** [comment]
Demo linking from the methodology age-adjustment page:

https://user-images.githubusercontent.com/41567007/159603092-5202ba44-795f-44af-9e64-e5be39ca54b3.mov

File: frontend/src/data/config/MetricConfig.ts [hunk 1]
```diff
@@ -38,30 +40,36 @@ export type MetricId =
   | "brfss_population_pct"
   | "copd_pct_share"
   | "copd_per_100k"
+  | "copd_ratio_age_adjusted"
   | "covid_cases"
```
Comment (joshzarrabi) [inline frontend/src/data/config/MetricConfig.ts:44]: @benhammondmusic Do we need age adjusted metrics for each variable?

File: frontend/src/data/config/MetricConfig.ts [hunk 2]
```diff
@@ -909,7 +1090,19 @@ export const METRIC_CONFIG: Record<DropdownVarId, VariableConfig[]> = {
...
+            "Age-Adjusted Voter Participation Ratio Compared to White (Non-Hispanic)",
+          shortLabel: "",
+          type: "ratio",
+        },
       },
     },
   ],
 };
+
+// export const AGE_ADJUSTED_VARIABLE_IDS: VariableId[] = [
```
Comment (joshzarrabi) [inline frontend/src/data/config/MetricConfig.ts:1105]: Delete this?

File: frontend/src/data/config/MetricConfig.ts (hunk 1 above)
Comment (benhammondmusic) [reply frontend/src/data/config/MetricConfig.ts:44]: Ya, the way everything is hooked together now the card looks for that property of metric id and I was getting a "something went wrong" error without them. I could look into getting the card to stop looking for that metric under certain conditions, but it seemed easier to just include the metric and have it be empty instead (since we are still displaying the card even if the metric isn't available)

Comment (joshzarrabi) [reply frontend/src/data/config/MetricConfig.ts:44]: No worries, that makes sense.

**Review by joshzarrabi — APPROVED** [review]
